### ISSUE TITLE
question/feature: how to use multiple state files and one tiller?

### ISSUE THREAD

**Issue by flypenguin** [Praqma/helmsman#173]
I use two DSFs, and if I execute helmsman it will delete the releases of the other DSF.
how can I prevent that while using the same tiller?

I have found the `--keep-untracked-releases` flag, but I would like to define a "scope" if you like about what helmsman should manage. so I can still just remove something from the DSF and it gets deleted, which is simple and convenient, while being able to use multiple state files. I think that is a benefit for coordination between devs as well.

Helmsman: v1.7.2

**Comment by sami-alajrami** [comment]
@flypenguin True. if you use the same Tiller from multiple DSFs, they will overlap and mess each other's work.
One way around this, is to manage each DSF apps with a unique Tiller. 

An ultimate fix is to add a customizable label to each DSF which is then used by Helmsman to identify the source of each deployed app. Currently, the label is always the same (not customizable) and is set to 'MANAGED_BY=HELMSMAN`  
But it's possible to make it something like `MANAGED_BY=HELMSMAN, DSF=DEV1` for instance.

**Comment by luisdavim** [comment]
The way I do it, is to have a common DSF and separate partial DSF for different contexts (ex. NameSpaces) and then call helmsman like this:
```
# Apply context 1
$ helmsman -f common.yaml -f context1.yaml
# Apply context 2
$ helmsman -f common.yaml -f context2.yaml
```

**Comment by flypenguin** [comment]
@sami-alajrami  This ultimate fix? That is something I would really like to see :)

@luisdavim hm. I guess this works, but I don't like it. I basically manage cert-manager and ingress-controller with helmsman, and having to include this in every helmsman call is just annoying to me. Thanks for sharing though! I might actually have to do it this way 🤷‍♂️

**Comment by mayppong** [comment]
Hmm since helmsman 3 will support "context" for the DSF, can this issue be resolved by the context?

**Comment by luisdavim** [comment]
Yes, the context does solve this issue.

**Comment by sami-alajrami** [comment]
this is fixed in helmsman v3.0.0 as indicated above. Closing this.
